eZ/Publish/API/Repository/Values/Content/Query/Criterion/RemoteId.php

```diff
@@ -23,13 +23,20 @@ class RemoteId extends Criterion
     /**
      * Creates a new remoteId criterion.
      *
-     * @param int|int[] $value One or more remoteId that must be matched
+     * @param int|int[]|string|string[] $value One or more remoteId that must be matched
      *
      * @throws \InvalidArgumentException if a non numeric id is given
      * @throws \InvalidArgumentException if the value type doesn't match the operator
      */
     public function __construct($value)
     {
+        // EZP-30502 Convert int to string to avoid database errors
+        if (\is_int($value)) {
+            $value = (string) $value;
```

Comment:
According to the criterion specifications,  type of `$value` could be int, string or list of one of those

https://github.com/ezsystems/ezpublish-kernel/blob/37d41f6651b8ca6408929d247d35b9ff71971898/eZ/Publish/API/Repository/Values/Content/Query/Criterion/RemoteId.php#L36-L50

IMHO this conversion should be handled in https://github.com/ezsystems/ezpublish-kernel/blob/e2d26faa54715675fc8f1eef3defbe192973bf5b/eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/RemoteId.php#L51-L54
 
WDYT @alongosz @andrerom?

Reply:
indeed